From 59a62dec8c6dd9761972adcf519ed9ed109c3dd4 Mon Sep 17 00:00:00 2001
From: Thomas Nizan <tnizan@witekio.com>
Date: Tue, 14 Jun 2022 13:21:37 +0000
Subject: [PATCH 1/2] iio: imu: inv_icm42600: fix device_get_match_data issue

device_get_match_data will return the type of chip (enum
inv_icm42600_chip) used in the device tree.

The chip number cannot be 0, as it will be considered as unknown
---
 drivers/iio/imu/inv_icm42600/inv_icm42600.h      | 4 +++-
 drivers/iio/imu/inv_icm42600/inv_icm42600_core.c | 2 +-
 2 files changed, 4 insertions(+), 2 deletions(-)

diff --git a/drivers/iio/imu/inv_icm42600/inv_icm42600.h b/drivers/iio/imu/inv_icm42600/inv_icm42600.h
index c0f5059b13b3..4968f21f0e51 100644
--- a/drivers/iio/imu/inv_icm42600/inv_icm42600.h
+++ b/drivers/iio/imu/inv_icm42600/inv_icm42600.h
@@ -17,10 +17,12 @@
 #include "inv_icm42600_buffer.h"
 
 enum inv_icm42600_chip {
-	INV_CHIP_ICM42600,
+	/* 0 is reserved to avoid clashing with NULL */
+	INV_CHIP_ICM42600 = 1,
 	INV_CHIP_ICM42602,
 	INV_CHIP_ICM42605,
 	INV_CHIP_ICM42622,
+	INV_CHIP_ICM42688,
 	INV_CHIP_NB,
 };
 
diff --git a/drivers/iio/imu/inv_icm42600/inv_icm42600_core.c b/drivers/iio/imu/inv_icm42600/inv_icm42600_core.c
index 8bd77185ccb7..8b719627df7d 100644
--- a/drivers/iio/imu/inv_icm42600/inv_icm42600_core.c
+++ b/drivers/iio/imu/inv_icm42600/inv_icm42600_core.c
@@ -565,7 +565,7 @@ int inv_icm42600_core_probe(struct regmap *regmap, int chip, int irq,
 	bool open_drain;
 	int ret;
 
-	if (chip < 0 || chip >= INV_CHIP_NB) {
+	if (chip < 1 || chip >= INV_CHIP_NB) {
 		dev_err(dev, "invalid chip = %d\n", chip);
 		return -ENODEV;
 	}
-- 
2.17.1

